query-planner-js/src/composedSchema/__tests__/removeInaccessibleElements.test.ts

```diff
@@ -148,6 +153,130 @@ describe('removeInaccessibleElements', () => {
     expect(schema.getType('Baz')).toBeDefined();
   });
 
+  it(`removes @inaccessible types from a union's types`, () => {
+    let schema = buildSchema(`
+      directive @core(feature: String!, as: String, for: core__Purpose) repeatable on SCHEMA
+
+      directive @inaccessible on FIELD_DEFINITION | OBJECT | INTERFACE | UNION
+
+      schema
+        @core(feature: "https://specs.apollo.dev/core/v0.2")
+        @core(feature: "https://specs.apollo.dev/inaccessible/v0.1")
+      {
+        query: Query
+      }
+
+      enum core__Purpose {
+        EXECUTION
+        SECURITY
+      }
+
+      type Query {
+        fooField: Foo @inaccessible
+      }
+
+      union Foo = Bar | Baz
+
+      type Bar @inaccessible {
+        someField: String
+      }
+
+      type Baz {
+        anotherField: String
+      }
+    `);
+
+    schema = removeInaccessibleElements(schema);
+
+    const fooType = schema.getType('Foo') as GraphQLUnionType;
+    expect(fooType).toBeDefined();
+    expect(fooType.getTypes()).toHaveLength(1);
```

Comment:
You may want to test for the right types more explicitly, perhaps `expect(fooType.getTypes().map(t => t.name)).toEqual(['Baz'])`?